From d2a05eebb3e19068c867b55ddd3ccd5a5b28ff03 Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Fri, 13 Sep 2024 10:12:50 +0300
Subject: [PATCH] [FSW-6580] fs1: use boot info set

With this patch, added support to use boot info
set to write boot info data over lfs. There is
no init, as it was taken care in lfs read/write.

Change-Id: I1d76bc8c8e98f099c8d0bcb6568d7df2033e4098
---

diff --git a/arch/fs1/arc/dev_boot_info.c b/arch/fs1/arc/dev_boot_info.c
index c9bfd67..f91c861 100644
--- a/arch/fs1/arc/dev_boot_info.c
+++ b/arch/fs1/arc/dev_boot_info.c
@@ -29,7 +29,7 @@
 	return false;
 }
 
-void boot_rom_init_boot_info(void)
+void boot_rom_set_boot_info(void)
 {
 	struct fs1_cold_rst_data cold_rst_data;
 
@@ -45,13 +45,6 @@
 		cold_rst_data.boot_info_set_done = 1;
 		regmap_writel(cold_rst_data.data, IO0_H0_PSOC_BOOT(mmPSOC_BOOT_CONF_COLD_RST_REG_2));
 		boot_rom_status_update(PROGRESS_BOOT_INFO_IMG_DESC_INFO_SET_FINISH);
-	} else {
-		/* Read and make sure databackup after any reset */
-		boot_rom_status_update(PROGRESS_BOOT_INFO_IMG_DESC_INFO_GET_START);
-
-		/* TODO: retrieve boot info and verify */
-
-		boot_rom_status_update(PROGRESS_BOOT_INFO_IMG_DESC_INFO_GET_FINISH);
 	}
 
 	return;
diff --git a/arch/fs1/arc/dev_boot_info.h b/arch/fs1/arc/dev_boot_info.h
index d056125..52da37a 100644
--- a/arch/fs1/arc/dev_boot_info.h
+++ b/arch/fs1/arc/dev_boot_info.h
@@ -13,4 +13,6 @@
 #define CONFIG_BOOTINFO_ADDR	(HL_FW_MEM_BASE + SCRATCHPAD_SRAM_OFFSET)
 #define CONFIG_BOOTINFO_SIZE	OFFSET_32KB
 
+void boot_rom_set_boot_info(void);
+
 #endif	/* _DEV_BOOT_INFO_H_ */
diff --git a/common/boot_flow.c b/common/boot_flow.c
index e671413..0d415d6 100644
--- a/common/boot_flow.c
+++ b/common/boot_flow.c
@@ -13,6 +13,7 @@
 #include <recovery.h>
 #include <boot_flow.h>
 #include <dev_uart.h>
+#include <dev_boot_info.h>
 
 /* These fields will be added to proper structure later */
 static enum hl_fw_manifest_type primary_manifest_idx = HL_FW_SPI_MANIFEST_0;
@@ -145,7 +146,7 @@
 
 	/* TODO: Set other manifest as invalid */
 
-	boot_rom_init_boot_info();
+	boot_rom_set_boot_info();
 
 	/* TODO: TPM PCR extend */
 
diff --git a/include/boot_rom.h b/include/boot_rom.h
index 3d82d68..e992548 100644
--- a/include/boot_rom.h
+++ b/include/boot_rom.h
@@ -127,7 +127,6 @@
 void boot_rom_wait_for_signal(void);
 void boot_rom_check_recovery(void);
 void boot_rom_bringup_ppboot(void);
-void boot_rom_init_boot_info(void);
 void boot_rom_tpm_spi_global_ctx_init(struct spi_ctx *tpm_spi);
 void boot_rom_tpm_spi_init(void);
 void boot_rom_tpm_spi_setup(void);
